README.mkd

```diff
@@ -61,6 +61,23 @@ The ISA subset naming conventions are described in Chapter 27 of the RISC-V
 user-level ISA specification. However, tools do not currently follow this 
 specification (input is case sensitive, ...).
 
+### Profile-based format
+
+Profiles should be recognized and used in the `-march=` option.
+The benefit is that it is easy for toolchain parsing the profiles string and expanding it into normal extensions combinations. It’s also easy to combine a profile with any extensions.
+
+As the spec defines, to use the profiles it should follow profile naming convention (See [3.4 form spec doc](https://github.com/riscv/riscv-profiles)), the toolchain will check whether an input profile name is correct at first, then do the parse  work.
+  
+In order to distinguish between ordinary extension input and input with profiles, profiles are assumed to be entered `at the beginning of the -march option`, and then input other extensions. Profiles `should use uppercase letters` in the -march option.
+
+e.g. `-march = RVA20U64` is a legal profile input, it will be expand into:
+
+`-march = rv64imafdc_zicsr_ziccif_ziccrse_ziccamoa_zicclsm_za128rs`
+which include all the mandatory extensions required by this profile.
+Here we are `not dealing with optional extensions` to profiles in the toolchain.
+
+Moreover, some extensions are still not ratified yet. So the toolchain only keeps their name strings only.
+
 If the 'C' (compressed) instruction set extension is targeted, the compiler 
 will generate compressed instructions where possible.
```

Comment:
Those two lines should move before `### Profile-based format`?

Reply:
Thanks for your metion, moved~